torchgeo/datasets/bigearthnet.py

```diff
@@ -573,3 +582,325 @@ def plot(
+
+        match self.bands:
+            case 's1':
+                sample['image'] = self._load_image(index, 's1')
+            case 's2':
+                sample['image'] = self._load_image(index, 's2')
+            case 'all':
+                sample['image_s1'] = self._load_image(index, 's1')
+                sample['image_s2'] = self._load_image(index, 's2')
+
+        sample['mask'] = self._load_map(index)
+        sample['label'] = self._load_target(index)
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_image(self, index: int, sensor: str) -> Tensor:
+        """Generic image loader for both S1 and S2.
+
+        Args:
+            index: index to return
+            sensor: 's1' or 's2'
+
+        Returns:
+            the sensor image
+        """
+        row = self.metadata_df.loc[index]
+        id_field = 's1_name' if sensor == 's1' else 'patch_id'
+        patch_id = row[id_field]
+        if sensor == 's2':
+            patch_dir = '_'.join(patch_id.split('_')[0:-2])
+        else:
+            patch_dir = '_'.join(patch_id.split('_')[0:-3])
+
+        paths = glob.glob(
+            os.path.join(
+                self.root,
+                self.metadata_locs[sensor]['directory'],
+                patch_dir,
+                patch_id,
+                '*.tif',
+            )
+        )
+
+        if sensor == 's2':
+            paths = sorted(paths, key=sort_sentinel2_bands)
+        else:
+            paths = sorted(paths)
+
+        images = []
+        for path in paths:
+            with rasterio.open(path) as dataset:
+                array = dataset.read(
+                    indexes=1,
+                    out_shape=self.image_size,
+                    out_dtype='int32',
+                    resampling=Resampling.bilinear,
+                )
+                images.append(array)
+
+        return torch.from_numpy(np.stack(images, axis=0)).float()
+
+    def _load_map(self, index: int) -> Tensor:
+        """Load a single image.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the Corine Land Cover map
+        """
+        row = self.metadata_df.loc[index]
+        patch_id = row['patch_id']
+        patch_dir = '_'.join(patch_id.split('_')[0:-2])
+        path = os.path.join(
+            self.root,
+            self.metadata_locs['maps']['directory'],
+            patch_dir,
+            patch_id,
+            patch_id + '_reference_map.tif',
+        )
+        with rasterio.open(path) as dataset:
+            map = dataset.read(out_dtype='int32')
+        return torch.from_numpy(map).long()
+
+    def _load_target(self, index: int) -> Tensor:
+        """Load the target mask for a single image.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the target label
+        """
+        indices = self.metadata_df.iloc[index]['labels']
```

Comment:
Also need to check the labels. In the V1 class it only seems possible to select the 19 label versions, because with selecting 43 they also get mapped to 19 if I understand correctly.